bot/internal/bot/docpaths.go

```diff
@@ -0,0 +1,122 @@
+package bot
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/gravitational/shared-workflows/bot/internal/github"
+	"github.com/gravitational/trace"
+)
+
+// DocsConfig represents the structure of the config.json file found in the docs
+// directory of a Teleport repo. We only use Redirects, so other fields have the
+// "any" type.
+type DocsConfig struct {
+	Navigation any
+	Variables  any
+	Redirects  RedirectConfig
+}
+
+type RedirectConfig []Redirect
+
+type Redirect struct {
+	Source      string
+	Destination string
+	Permanent   bool
+}
+
+// CheckDocsPathsForMissingRedirects takes the relative path to a
+// gravitational/teleport clone. It assumes that there is a file called
+// "docs/config.json" at the root of the directory that lists redirects in the
+// "redirects" field.
+//
+// CheckDocsPathsForMissingRedirects checks whether a PR has renamed or deleted
+// any docs files and, if so, returns an error if any of these docs files does
+// not correspond to a new redirect in the configuration file.
+func (b *Bot) CheckDocsPathsForMissingRedirects(ctx context.Context, teleportClonePath string) error {
+	if teleportClonePath == "" {
+		return trace.Wrap(errors.New("unable to load Teleport documentation config with an empty path"))
+	}
+
+	docsConfigPath := path.Join(teleportClonePath, "docs", "config.json")
+	f, err := os.Open(docsConfigPath)
+	if err != nil {
+		return trace.Wrap(fmt.Errorf("unable to open docs config file at %v: %v", docsConfigPath, err))
+	}
+	defer f.Close()
+
+	var c DocsConfig
+	if err := json.NewDecoder(f).Decode(&c); err != nil {
+		return trace.Wrap(fmt.Errorf("unable to load redirect configuration from %v: %v", docsConfigPath, err))
```

Comment:
Here and below, try to avoid creating two errors at once. The inner `fmt.Errorf` is rarely necessary.

```suggestion
		return trace.Wrap(err, "unable to load redirect configuration from %v: %v", docsConfigPath)
```